src/nvim/ex_cmds.c

```diff
@@ -4960,3 +4961,54 @@ void ex_oldfiles(exarg_T *eap)
     }
   }
 }
+
+void ex_trust(exarg_T *eap)
+{
+  char *p = skiptowhite(eap->arg);
+  size_t len = (size_t)(p - eap->arg);
+  char *arg1 = xmemdupz(eap->arg, len);
+
+  bool success = false;
+  bool action_allow = false, action_deny = false, action_remove = false;
+  char *msg = NULL;
+
+  if (len > 2 && STRNCMP(arg1, "++deny", len) == 0) {
+    char *path = skipwhite(p);
+    if (path[0] == '\0') {
+      path = NULL;
+    }
+    success = nlua_trust("deny", path, &msg);
+    action_deny = 1;
+  } else if (len > 2 && STRNCMP(arg1, "++remove", len) == 0) {
+    char *path = skipwhite(p);
+    if (path[0] == '\0') {
+      path = NULL;
+    }
+    success = nlua_trust("remove", path, &msg);
+    action_remove = 1;
+  } else {
+    char *path = skipwhite(eap->arg);
+    if (path[0] == '\0') {
+      path = NULL;
+    }
+    success = nlua_trust("allow", path, &msg);
+    action_allow = 1;
+  }
+
+  if (msg != NULL) {
+    if (success) {
+      if (action_allow) {
+        smsg("\"%s\" trusted.", msg);
+      } else if (action_deny) {
+        smsg("\"%s\" denied.", msg);
+      } else if (action_remove) {
+        smsg("\"%s\" removed.", msg);
+      }
+    } else {
+      semsg(e_trustfile, msg);
+    }
+  }
```

Comment:
We can save an allocation and a copy by printing this in `nlua_trust` instead of in `ex_trust` (then you can use the message while it's still on the Lua stack, before calling `lua_pop` without copying it).

Reply:
I was considering that but I wasn't sure if we wouldn't prefer to keep the wrapper "pure" by not having it print messages - I wasn't sure how much of a deal that was.
Will update :)

Reply:
Done in 7a93893914063e690a008b91552fbdb127c55876